tests/test_backends.py

```diff
@@ -1,25 +1,87 @@
 # Standard
+from unittest.mock import patch
+import pathlib
+import random
 import socket
+import tempfile
 
 # Third Party
 import pytest
 
 # First Party
 from instructlab.model.backends import backends
 
-supported_backends = ["llama-cpp", "vllm"]  # Example supported backends
-
-
-@pytest.fixture
-def mock_supported_backends(monkeypatch):
-    monkeypatch.setattr(
-        "instructlab.model.backends.backends.SUPPORTED_BACKENDS", supported_backends
-    )
-
 
 def test_free_port():
     host = "localhost"
     port = backends.free_tcp_ipv4_port(host)
     # check that port is bindable
     with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
         s.bind((host, port))
+
+
+# pylint: disable=redefined-outer-name
+def test_get_backend_auto_detection_fail_not_gguf():
+    with tempfile.NamedTemporaryFile() as temp_file:
```

Comment:
Use pytest fixture for temporary files:

```suggestion
def test_get_backend_auto_detection_fail_not_gguf(tmp_path: pathlib.Path):
    tmp_gguf = tmp_path / "test.gguf"
```